### PR TITLE
A simple bus to pipe out events including video recorded for analytics

### PR DESCRIPTION
**Pull request by prsauer**

### CODE REVIEW

File: packages/recorder/src/messageBus.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { EventEmitter } from 'events';
+
+import { VideoQueueItem } from './types';
+
+export class ManagerMessageBus extends EventEmitter {
```
Comment (atinylittleshell) [inline packages/recorder/src/messageBus.ts:5]: i think EventEmitter can accept generic type parameters to enforce strong type on what events are allowed

**Review by atinylittleshell — APPROVED** [review]

File: packages/recorder/src/messageBus.ts (hunk 1 above)
Comment (prsauer) [reply packages/recorder/src/messageBus.ts:5]: hrm I couldn't find anything in the built-in @types stuff... link to ref? It seemed on search that a lot of people rolled their own to do something like that

File: packages/recorder/src/messageBus.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { EventEmitter } from 'eventemitter3';
+
+import { VideoQueueItem } from './types';
+
+interface EventSpec {
+  'video-written': (data: VideoQueueItem) => void;
+}
+
+export class ManagerMessageBus extends EventEmitter<EventSpec> {}
```
Comment (atinylittleshell) [inline packages/recorder/src/messageBus.ts:9]: can't remember exactly but it might need to be something like EventEmitter<EventNames, EventArgs>

Comment (atinylittleshell) [reply packages/recorder/src/messageBus.ts:9]: where type EventNames = 'event1' | 'event2';

**Pull request merged**